src/internal/hooks/use-visual-mode/index.ts

```diff
@@ -11,10 +11,11 @@ import { warnOnce } from '../../logging';
 export function useCurrentMode(elementRef: React.RefObject<HTMLElement>) {
   const [value, setValue] = useState<'light' | 'dark'>('light');
   useMutationObserver(elementRef, node => {
-    const darkModeParent = findUpUntil(
-      node,
-      node => node.classList.contains('awsui-polaris-dark-mode') || node.classList.contains('awsui-dark-mode')
-    );
+    const darkModeParent =
+      findUpUntil(
+        node,
+        node => node.classList.contains('awsui-polaris-dark-mode') || node.classList.contains('awsui-dark-mode')
+      ) && !window?.matchMedia?.('print').matches;
```

Comment:
How about we will extract this as a separate function and reuse in `src/code-editor/util.ts`? It seems like it is now complex enough to not be duplicated

Also, can we write an extra test-case here? https://github.com/cloudscape-design/components/blob/main/src/internal/hooks/use-visual-mode/__tests__/use-visual-mode.test.tsx